### PR TITLE
Dependencies update and related changes

### PR DESCRIPTION
**Pull request by emartynov**
I got a question on SO how to make these test run again. So I tried to upgrade to latest dependencies and do some minor cleanups.

Please review and comment!

### CODE REVIEW

**Comment by jimandreas** [comment]
Your changes look quite good to me.   I have forked your version and have made follow-on changes to also clean things up a bit.  Good work!

**Comment by jimandreas** [comment]
I have looked into updating versions of various of the test libraries - but quickly hit issues
with Java 7 or Java 8 features that the newer versions reference.   Therefore I have tagged this
version as "works with AS231".   I am also seeing some issues when testing with AS24 previews
4 and 5 - but haven't managed to isolate just what is breaking with this upgrade.  Things are 
a bit fragile, it seems.   So for now I recommend sticking with AS231 and the tagged version.

**Comment by emartynov** [comment]
@jongerrish @erd, Is it ongoing project? Does our PR have chances to be merged?

**Comment by jongerrish** [comment]
Ya, I'll merge it when I get in tonight. :-)

**Comment by Zhuinden** [comment]
Yup

File: android-api-16/src/test/java/com/example/activity/MainActivityTest.java [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
-import org.robolectric.RobolectricGradleTestRunner;
+import org.robolectric.RobolectricTestRunner;
 import org.robolectric.annotation.Config;
 
 import static org.junit.Assert.assertTrue;
 
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class)
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 16)
```
Comment (jongerrish) [inline android-api-16/src/test/java/com/example/activity/MainActivityTest.java]: Build.VERSION_CODES.JELLYBEAN

File: android-api-16/build.gradle [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
 android {
-    compileSdkVersion 16
-    buildToolsVersion "21.1.2"
+    compileSdkVersion compileSdkVer
+    buildToolsVersion buildToolsVer
 
     defaultConfig {
-        applicationId "com.example"
+        applicationId 'com.example'
+        versionCode="2"
```
Comment (jongerrish) [inline android-api-16/build.gradle]: Is there a reason why you switch between single and double quotes?

File: android-api-16/build.gradle [hunk 3]
```diff
@@ -1,21 +1,31 @@
-apply plugin: "com.android.application"
+apply plugin: 'com.android.application'
 
 android {
```
Comment (jongerrish) [inline android-api-16/build.gradle]: You can include the new AGP integration as demoed in the Deckard project. Works with AS 3.0

https://github.com/robolectric/deckard/blob/master/build.gradle#L34

File: android-api-16/src/test/java/com/example/service/SampleIntentServiceTest.java [hunk 4]
```diff
@@ -7,23 +7,24 @@
...
 import org.robolectric.RuntimeEnvironment;
 import org.robolectric.annotation.Config;
 import org.robolectric.fakes.RoboSharedPreferences;
 
 import static org.junit.Assert.assertNotSame;
 
-@RunWith(RobolectricGradleTestRunner.class)
-@Config(constants = BuildConfig.class)
+@RunWith(RobolectricTestRunner.class)
+@Config(constants = BuildConfig.class, sdk = 16)
```
Comment (jongerrish) [inline android-api-16/src/test/java/com/example/service/SampleIntentServiceTest.java]: same - Build.VERSION_CODES.JELLYBEAN apply throughout.

File: android-api-16/build.gradle (hunk 3 above)
Comment (jongerrish) [reply android-api-16/build.gradle]: Apply to other gradle build files too

**Pull request closed without merging**